packages/containers/src/SelectObject/SelectObject.container.js

```diff
@@ -69,29 +79,74 @@ class SelectObject extends React.Component {
 		sourceData: PropTypes.array,
 		selectedId: PropTypes.string,
 		tree: PropTypes.object,
+		idAttr: PropTypes.string,
+		nameAttr: PropTypes.string,
+		breadCrumbsRootLabel: PropTypes.string,
 	};
 	static defaultProps = {
 		sourceData: [],
+		idAttr: 'id',
+		nameAttr: 'name',
+		breadCrumbsRootLabel: 'root',
 	};
 
 	constructor(props) {
 		super(props);
 		this.state = {};
 		this.filter = filter;
 		this.getById = getById;
+		this.onKeyDown = this.onKeyDown.bind(this);
+		this.onTreeClick = this.onTreeClick.bind(this);
+		this.onResultsClick = this.onResultsClick.bind(this);
+	}
+
+	onKeyDown(event) {
+		// handle arrows only if query has been set
+		if (event.keyCode === keycode.codes.down) {
+			// put focus on treeview or results
+			this.props.setState({ focus: 'content' });
+		}
+	}
+
+	onFilterFocus() {
+		this.props.setState({ focus: 'FilterBar' });
+	}
+
+	onTreeClick(data) {
+		this.props.setState({ selectedId: data[this.props.idAttr], focus: 'content' });
+	}
+
+	onResultsClick(event, item) {
+		this.props.setState({ selectedId: item.get(this.props.idAttr) });
 	}
 
 	render() {
+		const state = this.props.state || DEFAULT_STATE;
 		const props = omit(this.props, cmfConnect.INJECTED_PROPS);
 		const matches = [];
 		function addMatch(item) {
 			matches.push(item);
 		}
-		props.filteredData = this.filter(props.sourceData, props.query, {
-			...props,
-			onMatch: addMatch,
-		});
-		let selected = props.selectedId;
+		let selected = state.get('selectedId') || props.selectedId;
+		if (props.query) {
```

Comment:
5 `if` statements on this render function im sure it can be cleaned up a little bit